### PR TITLE
fix: Update flight model info in fmgc guidance

### PR DESCRIPTION
**Pull request by donstim**
Fixes #[issue_no]

## Summary of Changes

This change updates the flight model information in the FMGC guidance section for use in computing vertical guidance parameters. This change will make the flight model information in the FMGC consistent with https://github.com/flybywiresim/a32nx/pull/6820.
## Additional context

Discord username (if different from GitHub): donbikes#4084

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by beheh — APPROVED** [review]
I'm approving this from a technical side, because the form of the changes seem fine. I haven't cross-checked the numbers themselves.

**Comment by davidwalschots** [comment]
@donstim What kind of testing are you expecting on this item?

**Comment by donstim** [comment]
> @donstim What kind of testing are you expecting on this item?

Fair question, but I don't have an answer. This PR was done as a courtesy to @BlueberryKing and anyone else who might be working on FMGC guidance that needs this information. I can provide some testing guidance for anyone who can access the `baseDrag` value computed by this file by inputting a CL value, but I don't think that would be doable by a typical QA tester. I would envision a 2-step test: 1) Ensure the computed `baseDrag` value matches values I can provide that were used to create the polynomials in the equations; and 2) Verify that the computed `baseDrag` value matches (within a tolerance) CD values shown in the Aircraft Editor Debug Speeds window in developer mode for any given flight condition.

**Comment by tracernz** [comment]
I think this is one that really just needs review, although maybe @BlueberryKing can do some tests with his vnav branch.

**Comment by BlueberryKing** [comment]
I have been using this PR on the `experimental-vnav` branch. As expected, I have not noticed any breaking issues with it, which is the important part before merging this into master.

I would be happy to do a more comprehensive test on the functionality using the method @donstim described.

**Comment by tracernz** [comment]
I am inclined to just merge this. Any objections?

**Comment by BlueberryKing** [comment]
> I am inclined to just merge this. Any objections?

Yeah, good to go from my side. Has not caused any issues on experimental.

**Comment by donstim** [comment]
Sorry for the delay. Finally getting back to this. Here are some inputs/outputs that can be used as a check:

FLAPS UP
CL = 0.29471, BaseDrag = 0.0213
For Mach = 0.82, CD Mach correction = 0.00122, CD with Mach correction = 0.02252

CL = 0.39013, BaseDrag = 0.023299
For Mach = 0.65, CD Mach correction = 0.0004, CD with Mach correction = 0.023699

CONF 1
CL = 0.84155, BaseDrag = 0.055432

CONF 1+F
CL = 1.07326, BaseDrag = 0.076367

CONF 2
CL = 1.00455, BaseDrag = 0.0.79944

CONF 3
CL = 1.07352, BaseDrag = 0.088355,

CONF FULL
CL = 1.22098, BaseDrag = 0.118748

**Comment by BlueberryKing** [comment]
Thank you very much! I will be checking these values.

**Comment by BlueberryKing** [comment]
I have now conducted the tests, both with the values @donstim provided as well as with the computed values in the simulator. The values lined up nicely for all configurations. 👍

**Comment by donstim** [comment]
Someone needs to add the Mach correction to BaseDrag. Otherwise, it will not be very accurate for flaps up high speed drag
Screenshot (1994)
.

**Comment by BlueberryKing** [comment]
I have now added the Mach correction factor into the flight model predictions. I tested this briefly and observed more conservative predictions, as would be expected since we now correctly account for an increase in drag at high Mach numbers.

In the images below we can see how the top of climb arrow moves further away with the Mach correction.

Without the Mach correction:
Screenshot 2022-03-20 131330

With Mach correction:
Screenshot 2022-03-20 131810

I reached my cruising altitude closer to the predicted top of climb with these changes. Good stuff!

File: src/fmgc/src/guidance/vnav/FlightModel.ts [hunk 1]
```diff
@@ -83,7 +88,37 @@ export class FlightModel {
...
+     * @param xs The table of x values with known y values
+     * @param ys The y values corresponding to the x values in `xs`
+     */
+    static interpolate(x: number, xs: number[], ys: number[]) {
+        if (x <= xs[0]) {
+            return ys[0];
+        }
+
+        for (let i = 0; i < xs.length - 1; i++) {
+            if (x > xs[i] && x < xs[i + 1]) {
```
Comment (tracernz) [inline src/fmgc/src/guidance/vnav/FlightModel.ts:116]: Shouldn’t one of these be or equal?

**Review by tracernz — APPROVED** [review]

File: src/fmgc/src/guidance/vnav/FlightModel.ts (hunk 1 above)
Comment (BlueberryKing) [reply src/fmgc/src/guidance/vnav/FlightModel.ts]: Absolutely right, had just noticed that. 😄

**Pull request merged**